### DISCUSSION TITLE
why not use tilingsprite

### DISCUSSION THREAD

**Discussion by ChasLui** [jasonsturges/pixi-graphpaper#5] [Q&A]
From the official description, the use of tilingsprite can bring better performance.

**Comment by jasonsturges** [comment]
@ChasLui Hi, I find TilingSprite inaccurate the sub pixel level, having to `x %= width`.

The other issue is that major, intermediate, and minor rule are independent instead of subdivisions of each other.  

Therefore, the tile texture would have to be a common denominator of the three - quite large.

![graph-rule](https://user-images.githubusercontent.com/1213591/153937568-cee2f586-5f58-4f00-8ba7-b621c8c4235f.gif)

While this made sense for the applications and data models this was used for, it is contradictory to axis subdivisions in things like my [Pixi Gauges](https://github.com/jasonsturges/pixi-gauges) components.

The concept here is more pixel-level artboard and authoring, whereas the axis implementations presume range is calculated.

If this project adapted that philosophy, tiling would make more sense.

In this example, that would be:

- major grid size = view width / 100
- intermediate grid size = major width / 50, or 2 ticks per major grid
- minor grid size = 
  - major width / 10, or 10 tickets per major grid size
  - intermediate width / 5, or 5 tickets per intermediate grid size

Been thinking about an autocalc for this library - I might revisit that if I get a chance to return to my authoring project.

**Comment by ChasLui** [comment]
> @ChasLui Hi, I find TilingSprite inaccurate the sub pixel level, having to `x %= width`.
> 
> The other issue is that major, intermediate, and minor rule are independent instead of subdivisions of each other.
> 
> Therefore, the tile texture would have to be a common denominator of the three - quite large.
> 
> ![graph-rule](https://user-images.githubusercontent.com/1213591/153937568-cee2f586-5f58-4f00-8ba7-b621c8c4235f.gif)
> 
> While this made sense for the applications and data models this was used for, it is contradictory to axis subdivisions in things like my [Pixi Gauges](https://github.com/jasonsturges/pixi-gauges) components.
> 
> The concept here is more pixel-level artboard and authoring, whereas the axis implementations presume range is calculated.
> 
> If this project adapted that philosophy, tiling would make more sense.
> 
> In this example, that would be:
> 
> * major grid size = view width / 100
> * intermediate grid size = major width / 50, or 2 ticks per major grid
> * minor grid size =
>   
>   * major width / 10, or 10 tickets per major grid size
>   * intermediate width / 5, or 5 tickets per intermediate grid size
> 
> Been thinking about an autocalc for this library - I might revisit that if I get a chance to return to my authoring project.

Thanks for answering questions
